lib/spack/spack/install_test.py

```diff
@@ -52,6 +94,640 @@ def get_test_stage_dir():
     )
 
 
+def cache_extra_test_sources(pkg: Pb, srcs: ListOrStringType):
+    """Copy relative source paths to the corresponding install test subdir
+
+    This routine is intended as an optional install test setup helper for
+    grabbing source files/directories during the installation process and
+    copying them to the installation test subdirectory for subsequent use
+    during install testing.
+
+    Args:
+        pkg: package being tested
+        srcs: relative path for file(s) and or subdirectory(ies) located in
+            the staged source path that are to be copied to the corresponding
+            location(s) under the install testing directory.
+
+    Raises:
+        OSError: if any of the source paths do not exist under the build stage
+        ValueError: if any of the source paths are absolute
+    """
+    paths = [srcs] if isinstance(srcs, str) else srcs
+    if any(os.path.isabs(path) for path in paths):
+        raise ValueError("Only relative paths can be used to copy test sources")
```

Comment:
(Also for later) I'm a bit worried about raising built-in exception like `OSError` We might turn this into a `SpackError` of some kind in the future.

Reply:
IIRC I used `OSError` because that is what would have been returned by the `fs` call if the path didn't exist.  I was just trying to avoid the extra calls when I could check in advance.

Reply:
Changed to capture and report the errors for the method.